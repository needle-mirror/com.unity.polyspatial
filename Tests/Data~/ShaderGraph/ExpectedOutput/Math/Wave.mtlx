<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <dot name = "_1" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_2" />
    </dot>
    <combine4 name = "_2" type = "vector4">
        <input name = "in1" type = "float" nodename = "_3" />
        <input name = "in2" type = "float" nodename = "_4" />
        <input name = "in3" type = "float" nodename = "_5" />
        <input name = "in4" type = "float" nodename = "_6" />
    </combine4>
    <dot name = "_3" type = "float">
        <input name = "in" type = "float" nodename = "_11" />
    </dot>
    <dot name = "_4" type = "float">
        <input name = "in" type = "float" nodename = "_17" />
    </dot>
    <dot name = "_5" type = "float">
        <input name = "in" type = "float" nodename = "_23" />
    </dot>
    <dot name = "_6" type = "float">
        <input name = "in" type = "float" value = "0" />
    </dot>
    <dot name = "_7" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_8" />
    </dot>
    <combine3 name = "_8" type = "vector3">
        <input name = "in1" type = "float" nodename = "_3" />
        <input name = "in2" type = "float" nodename = "_4" />
        <input name = "in3" type = "float" nodename = "_5" />
    </combine3>
    <dot name = "_9" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_10" />
    </dot>
    <combine2 name = "_10" type = "vector2">
        <input name = "in1" type = "float" nodename = "_3" />
        <input name = "in2" type = "float" nodename = "_4" />
    </combine2>
    <dot name = "_11" type = "float">
        <input name = "in" type = "float" nodename = "_12" />
    </dot>
    <subtract name = "_12" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_13" />
    </subtract>
    <multiply name = "_13" type = "float">
        <input name = "in1" type = "float" value = "2" />
        <input name = "in2" type = "float" nodename = "_14" />
    </multiply>
    <round name = "_14" type = "float">
        <input name = "in" type = "float" nodename = "_15" />
    </round>
    <realitykit_fractional name = "_15" type = "float">
        <input name = "in" type = "float" nodename = "_16" />
    </realitykit_fractional>
    <dot name = "_16" type = "float">
        <input name = "in" type = "float" nodename = "_38" />
    </dot>
    <dot name = "_17" type = "float">
        <input name = "in" type = "float" nodename = "_18" />
    </dot>
    <multiply name = "_18" type = "float">
        <input name = "in1" type = "float" value = "2" />
        <input name = "in2" type = "float" nodename = "_19" />
    </multiply>
    <subtract name = "_19" type = "float">
        <input name = "in1" type = "float" nodename = "_20" />
        <input name = "in2" type = "float" nodename = "_21" />
    </subtract>
    <dot name = "_20" type = "float">
        <input name = "in" type = "float" nodename = "_38" />
    </dot>
    <floor name = "_21" type = "float">
        <input name = "in" type = "float" nodename = "_22" />
    </floor>
    <add name = "_22" type = "float">
        <input name = "in1" type = "float" value = "0.5" />
        <input name = "in2" type = "float" nodename = "_20" />
    </add>
    <dot name = "_23" type = "float">
        <input name = "in" type = "float" nodename = "_24" />
    </dot>
    <add name = "_24" type = "float">
        <input name = "in1" type = "float" nodename = "_25" />
        <input name = "in2" type = "float" nodename = "_27" />
    </add>
    <sin name = "_25" type = "float">
        <input name = "in" type = "float" nodename = "_26" />
    </sin>
    <dot name = "_26" type = "float">
        <input name = "in" type = "float" nodename = "_38" />
    </dot>
    <mix name = "_27" type = "float">
        <input name = "bg" type = "float" nodename = "_28" />
        <input name = "fg" type = "float" nodename = "_30" />
        <input name = "mix" type = "float" nodename = "_31" />
    </mix>
    <swizzle name = "_28" type = "float">
        <input name = "in" type = "vector2" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <dot name = "_29" type = "vector2">
        <input name = "in" type = "vector2" value = "-0.5,0.5" />
    </dot>
    <swizzle name = "_30" type = "float">
        <input name = "in" type = "vector2" nodename = "_29" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <realitykit_fractional name = "_31" type = "float">
        <input name = "in" type = "float" nodename = "_32" />
    </realitykit_fractional>
    <multiply name = "_32" type = "float">
        <input name = "in1" type = "float" nodename = "_33" />
        <input name = "in2" type = "float" value = "43758.55" />
    </multiply>
    <sin name = "_33" type = "float">
        <input name = "in" type = "float" nodename = "_34" />
    </sin>
    <multiply name = "_34" type = "float">
        <input name = "in1" type = "float" nodename = "_35" />
        <input name = "in2" type = "float" value = "91.2228" />
    </multiply>
    <subtract name = "_35" type = "float">
        <input name = "in1" type = "float" nodename = "_25" />
        <input name = "in2" type = "float" nodename = "_36" />
    </subtract>
    <sin name = "_36" type = "float">
        <input name = "in" type = "float" nodename = "_37" />
    </sin>
    <add name = "_37" type = "float">
        <input name = "in1" type = "float" nodename = "_26" />
        <input name = "in2" type = "float" value = "1" />
    </add>
    <constant name = "_Time" type = "vector4">
        <input name = "value" type = "vector4" value = "0,0,0,0" />
    </constant>
    <swizzle name = "_38" type = "float">
        <input name = "channels" type = "string" value = "y" />
        <input name = "in" type = "vector4" nodename = "_Time" />
    </swizzle>
    <GeometryModification name = "_39" type = "vertexshader">
    </GeometryModification>
    <realitykit_unlit name = "_40" type = "surfaceshader">
        <input name = "applyPostProcessToneMap" type = "boolean" value = "false" />
        <input name = "color" type = "color3" nodename = "_41" />
    </realitykit_unlit>
    <surfacematerial name = "USD_Wave" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_39" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_40" />
    </surfacematerial>
    <swizzle name = "_41" type = "color3">
        <input name = "in" type = "vector3" nodename = "_7" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
</materialx>
