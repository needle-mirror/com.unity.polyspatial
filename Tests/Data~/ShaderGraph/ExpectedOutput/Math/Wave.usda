#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Wave"
        {
            float4 inputs:_Time = (0, 0, 0, 0)

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_41.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_41"
            {
                uniform token info:id = "ND_swizzle_vector3_color3"
                float3 inputs:in.connect = <../_7.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_8.outputs:out>
                float3 outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_combine3_vector3"
                float inputs:in1.connect = <../_3.outputs:out>
                float inputs:in2.connect = <../_4.outputs:out>
                float inputs:in3.connect = <../_5.outputs:out>
                float3 outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_11.outputs:out>
                float outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_12.outputs:out>
                float outputs:out
            }

            def Shader "_12"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_13.outputs:out>
                float outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1 = 2
                float inputs:in2.connect = <../_14.outputs:out>
                float outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_round_float"
                float inputs:in.connect = <../_15.outputs:out>
                float outputs:out
            }

            def Shader "_15"
            {
                uniform token info:id = "ND_realitykit_fractional_float"
                float inputs:in.connect = <../_16.outputs:out>
                float outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_38.outputs:out>
                float outputs:out
            }

            def Shader "_38"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                string inputs:channels = "y"
                float4 inputs:in.connect = <../.inputs:_Time>
                float outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_17.outputs:out>
                float outputs:out
            }

            def Shader "_17"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_18"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1 = 2
                float inputs:in2.connect = <../_19.outputs:out>
                float outputs:out
            }

            def Shader "_19"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_20.outputs:out>
                float inputs:in2.connect = <../_21.outputs:out>
                float outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_floor_float"
                float inputs:in.connect = <../_22.outputs:out>
                float outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1 = 0.5
                float inputs:in2.connect = <../_20.outputs:out>
                float outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_38.outputs:out>
                float outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_23.outputs:out>
                float outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_24.outputs:out>
                float outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_25.outputs:out>
                float inputs:in2.connect = <../_27.outputs:out>
                float outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_sin_float"
                float inputs:in.connect = <../_26.outputs:out>
                float outputs:out
            }

            def Shader "_26"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_38.outputs:out>
                float outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_mix_float"
                float inputs:bg.connect = <../_28.outputs:out>
                float inputs:fg.connect = <../_30.outputs:out>
                float inputs:mix.connect = <../_31.outputs:out>
                float outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_29.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in = (-0.5, 0.5)
                float2 outputs:out
            }

            def Shader "_30"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_29.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }

            def Shader "_31"
            {
                uniform token info:id = "ND_realitykit_fractional_float"
                float inputs:in.connect = <../_32.outputs:out>
                float outputs:out
            }

            def Shader "_32"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_33.outputs:out>
                float inputs:in2 = 43758.55
                float outputs:out
            }

            def Shader "_33"
            {
                uniform token info:id = "ND_sin_float"
                float inputs:in.connect = <../_34.outputs:out>
                float outputs:out
            }

            def Shader "_34"
            {
                uniform token info:id = "ND_multiply_float"
                float inputs:in1.connect = <../_35.outputs:out>
                float inputs:in2 = 91.2228
                float outputs:out
            }

            def Shader "_35"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1.connect = <../_25.outputs:out>
                float inputs:in2.connect = <../_36.outputs:out>
                float outputs:out
            }

            def Shader "_36"
            {
                uniform token info:id = "ND_sin_float"
                float inputs:in.connect = <../_37.outputs:out>
                float outputs:out
            }

            def Shader "_37"
            {
                uniform token info:id = "ND_add_float"
                float inputs:in1.connect = <../_26.outputs:out>
                float inputs:in2 = 1
                float outputs:out
            }
        }
    }
}
