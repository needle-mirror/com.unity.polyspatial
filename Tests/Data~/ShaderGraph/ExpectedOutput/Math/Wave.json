{"properties":[{"name":"_Time","type":3}]}